### PR TITLE
S5

### PR DESCRIPTION
**Pull request by bentleybiscuit**
assuming matchedCards exists...

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -3,6 +3,7 @@
 <head>
     <script type="text/javascript" src="functions/displayImage.js" defer></script>
     <script type="text/javascript" src="functions/shuffle.js" defer></script>
+    <script type="text/javascript" src="functions/matchedCards.js" defer></script>
```
Comment (sudokufan) [inline index.html]: needs another line for checkMatch.js

File: functions/checkMatch.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
+/** check if two cards match **/
+/* This function:
+*       1. receives an array of all of the card's locations
+*       2. receives an array of chosen card locations
+*       3. uses the values in the chosen card array to determine which locations to check in the cards array
+*       4. gets the values stored in those locations in the card array
+*       5. checks if the two values in the card array match
+*       6. returns true if both match
+*       7. returns false if they do not match
```
Comment (sudokufan) [inline functions/checkMatch.js]: I'm not sure we need a summary here AND line-by-line comments throughout the function as well?

File: functions/checkMatch.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+*
+* */
+function checkMatch(cardsArray, chosenCards) {
+    //pick a random card from the array
+    //get the value of the first chosenCard and use it to decide the position to check in the cardsArray and store the result in a variable
+    var deckCardOne = cardsArray[chosenCards[0]];
+
+    //picks a random card for the second card
+    //get the value of the first chosenCard and use it to decide the position to check in the cardsArray and store the result in a variable
+    var deckCardTwo = cardsArray[chosenCards[1]];
```
Comment (sudokufan) [inline functions/checkMatch.js]: have a feeling these should be lets

File: functions/checkMatch.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+    //check if the two cards are strictly equal
+
+    if (deckCardOne === deckCardTwo){//yes
+        //return a boolean value of true
+        game.cardsTurn = []
+        //adding matched cards to game.matchedCards
+        if(game.matchedCards.length == 16) {
+            alert("Geckcellent work my friend! Wanna geckgo again?")
+        }
```
Comment (sudokufan) [inline functions/checkMatch.js]: 👏🏻

File: functions/checkMatch.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    // Getting the cards image numbers from the cards array
+    let card1Image = game.cardImageArray[card1Id]
+    let card2Image = game.cardImageArray[card2Id]
+
+    // Checking if the two cards images match
+    if (card1Image === card2Image) {
+        // Adds card ID's to the matched cards array
+        game.matchedCards.push(card1Id,card2Id)
+
+        game.cardsTurn.forEach(function (card) {
```
Comment (ashleycoles) [inline functions/checkMatch.js]: Remove space between function and params

File: functions/checkMatch.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+            // Adds matched class to matching cards
+            document.getElementById(card).classList = "card matched"
+            // Remove event listener
+            if (game.matchedCards.length == 16) {
+                setTimeout(function () {
```
Comment (ashleycoles) [inline functions/checkMatch.js]: Remove space between function and params

File: functions/checkMatch.js [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+                    alert("Geckcellent work! Want to geckgo again?")
+                }, 250)
+            }
+
+        })
+        
+    } else {
+        game.cardsTurn.forEach(function (card) {
```
Comment (ashleycoles) [inline functions/checkMatch.js]: Remove space between function and params

File: functions/checkMatch.js [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+            // Adds delay to resetting cards
+            setTimeout(function () {
```
Comment (ashleycoles) [inline functions/checkMatch.js]: Remove space between function and params

File: functions/checkMatch.js [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+    let card2Image = game.cardImageArray[card2Id]
+
+    // Checking if the two cards images match
+    if (card1Image === card2Image) {
+        // Adds card ID's to the matched cards array
+        game.matchedCards.push(card1Id,card2Id)
+
+        game.cardsTurn.forEach(function (card) {
+            // Adds matched class to matching cards
+            document.getElementById(card).classList = "card matched"
```
Comment (ashleycoles) [inline functions/checkMatch.js]: Use classList.add isntead (as per S4)

File: main.js [hunk 10]
```diff
@@ -1,36 +1,66 @@
...
+    turn: 0
 }
 
-game.cardArray = shuffle(game.cardArray)
+// Shuffles the image numbers
+game.cardImageArray = shuffle(game.cardImageArray)
 
-document.querySelector('.newGame').addEventListener('click', function(e) {
+// Event listener on the new game button that will reset the game 
+document.querySelector(".newGame").addEventListener("click", function (e) {
```
Comment (ashleycoles) [inline main.js]: Remove space between function and params

File: main.js [hunk 11]
```diff
@@ -1,36 +1,66 @@
...
-    let cards = document.querySelectorAll('.card')
-    cards.forEach(function(card) {
+
+    // Reset game state
+    game.matchedCards = []
+    game.clicks = 0
+    game.cardsTurn = []
+
+    // Turn cards back over
+    game.cards.forEach(function (card) {
```
Comment (ashleycoles) [inline main.js]: Remove space between function and params

File: main.js [hunk 12]
```diff
@@ -1,36 +1,66 @@
...
+    game.cardImageArray = shuffle(game.cardImageArray)
 })
 
-document.querySelector('.instructionsButton').addEventListener('click', function(e){
-    let instructions = document.querySelector('.instructions')
-    if (instructions.style.display == "none") {
-        instructions.style.display = "block"
+
+// Instructions button event listener
+document.querySelector(".instructionsButton").addEventListener("click", function (e) {
```
Comment (ashleycoles) [inline main.js]: Remove space between function and params

File: main.js [hunk 13]
```diff
@@ -1,36 +1,66 @@
...
-        instructions.style.display = "none"
+        instructions.classList.add("hidden")
     }
 })
 
 //forEach loop that calls displayImage function when a card is clicked
-game.cards.forEach(function(card) {
-    card.addEventListener('click', function(event){
+game.cards.forEach(function (card) {
+    card.addEventListener("click", function (event) {
```
Comment (ashleycoles) [inline main.js]: Remove space between function and params

File: main.js [hunk 14]
```diff
@@ -1,36 +1,66 @@
...
+    cardImageArray: [1, 1, 2, 2, 3, 3, 4, 4, 5, 5, 6, 6, 7, 7, 8, 8],
+    // Node list of all the card elements
+    cards: document.querySelectorAll(".card"),
+    // Array of the ID's of cards that have been matched
+    matchedCards: [],
+    // Stores the card ID's of the cards active in the current turn
+    cardsTurn: [],
+    // Stores the number of clicks in the current turn 0 - 2
+    clicks: 0,
+    turn: 0
```
Comment (ashleycoles) [inline main.js]: This doesn't reset when you hit new game.

File: functions/checkMatch.js [hunk 15]
```diff
@@ -0,0 +1,49 @@
...
+            // Remove event listener
+            if (game.matchedCards.length == 16) {
+                setTimeout(function () {
+                    alert("Geckcellent work! Want to geckgo again?")
```
Comment (ashleycoles) [inline functions/checkMatch.js]: This alert seems to pop up twice.

**Pull request merged**